docs/src/main/tut/typeclasses/comonad.md

```diff
@@ -0,0 +1,87 @@
+---
+layout: docs
+title:  "Comonad"
+section: "typeclasses"
+source: "core/src/main/scala/cats/Comonad.scala"
+scaladoc: "#cats.Comonad"
+---
+# Comonad
+
+Given any type class we can create a "Co" type class for which its functions
+are the "dual" of the other. A dual to a function has the same types but the 
+direction of the arrows are reversed. Whether or not that is useful even possible 
+depends on the particular type.
```

Comment:
It looks like here you try to simplify duality in the context of software programming. I feel like I need to see more justification for this simplification. Do you have a source for it or did you come up with it yourself?

Reply:
I came up with it myself based on various sources :) Perhaps it could be more rigorous

Reply:
I feel it's probably safer to avoid generalizing duality for "any type class" in software programming.
How about we just use the sentence below is to start and add your description of dual function after? 
And maybe add something like "For more formal definition of duality, please refer to https://ncatlab.org/nlab/show/duality."

```
`Comonad` is a `Functor` and provides duals of the [`Monad`](monad.html) `pure`
and `flatMap` functions.  A dual to a function has the same types but the 
direction of the arrows are reversed. Whether or not that is useful even possible 
depends on the particular type. For a more formal definition of duality, please refer to https://ncatlab.org/nlab/show/duality.
```